src/axom/sidre/nanobind_sidre.cpp

```diff
@@ -122,15 +122,55 @@ nb::ndarray<nb::numpy> bufferToNumpyArray(Buffer& self)
     /* dtype = */ typeIDToDtype(id));
 }
 
-// TODO determine return value policy (nb::rv_policy::reference or not) for ownership between C++ and Python:
-// https://nanobind.readthedocs.io/en/latest/ownership.html#return-value-policies
+/*!
+ * \brief Helper function to bind iterator types
+ *
+ * \tparam IteratorType The type of the iterator
+ */
+template <typename IteratorType>
+void bindIterator(nb::module_& m, const char* iterator_name)
+{
+  // Create non-const and const python iterators using nanobind's make_iterator helper
+  // We need to specify the ValueType (4th template parameter) as a reference since IndexedCollections
+  // have pointer/reference semantics and the underlying type (e.g. sidre::View) are not copyable
+  nb::class_<IteratorType>(m, iterator_name)
+    .def("__len__", &IteratorType::size)
+    .def(
+      "__iter__",
+      [iterator_name](IteratorType& self) {
+        return nb::make_iterator<nb::rv_policy::reference,
+                                 decltype(self.begin()),
+                                 decltype(self.end()),
+                                 typename IteratorType::CollectionType::value_type&>(
+          nb::type<IteratorType>(),
+          iterator_name,
+          self.begin(),
+          self.end());
+      },
+      nb::keep_alive<0, 1>())
+    .def(
+      "__iter__",
+      [iterator_name](const IteratorType& self) {
+        return nb::make_iterator<nb::rv_policy::reference,
+                                 decltype(self.cbegin()),
+                                 decltype(self.cend()),
+                                 const typename IteratorType::CollectionType::value_type&>(
+          nb::type<const IteratorType>(),
+          iterator_name,
```

Comment:
I'm not sure if this name for the const iterator has to be different from the name of the non-const one.

(I'm also not sure if this `const` version will ever get called from Python, so perhaps we should remove it ?)